scripts/src/modules/analytics/search/search_filters_tracking.js

```diff
@@ -1,21 +1,24 @@
 import push_to_data_layer from "./../push_to_data_layer";
 
-const getFilters = () => {
+const pushActiveFilterDataOnLoad = () => {
     // get filters after DOM has loaded and they have rendered on page
     window.addEventListener('load', () => {
-        const filters = document.querySelectorAll('[data-filter]');
-
-        loopFilters(filters);
+        const filterList = document.querySelector('.search-results__selected-filters')
+        if(filterList !== null) {
+            pushActiveFilterData(filterList);
+        }
     });
 }
 
-const loopFilters = (filters) => {
+const pushActiveFilterData = (filterList) => {
     // get current search bucket
-    const searchBucket = document.querySelector('[data-search-bucket]').getAttribute('data-search-bucket');
-    
+    const searchBucket = filterList.getAttribute('data-search-bucket');
+
+    const filters = filterList.querySelector("[data-filter]");
+
```

Comment:
Note / Suggestion: Depending on how defensive we want to be here. We could check that `searchBucket` / `filters` are not null values. Similar to how you've done above and log an error to the console. 

In theory, this should always have those values but I'll leave that choice to you. As you've pointed out this code is a little awkward anyway. It would be worth adding Sentry to the FE if we can to avoid this issue being overlooked in the future as I assume that's the case.

Reply:
Ahah. I was originally replying to say `querySelector()` will give me an empty list when there are no matches, but I was totally wrong about that. `querySelectorAll()` is what I really want here (there are supposed to be multiple matching elements).

Reply:
In regards to adding Sentry... I think that's definitely worthwhile, but probably one for a separate PR.